pkg/apis/core/types.go

```diff
@@ -650,6 +650,13 @@ type ISCSIVolumeSource struct {
 	// <target portal>:<volume name> will be created for the connection.
 	// +optional
 	InitiatorName *string
+	// iSCSI Replacement Timeout field required to set the iSCSI TimeoutSetting to wait for session re-establishment
+	// before failing SCSI commands back to the application when running the Linux SCSI Layer error handler.
+	// If the ReplacementTimeout value is 0, IO will be failed immediately.
+	// If the ReplacementTImeout value is less than 0, IO will remain queued until the session
+	// is logged back in, or until the user runs the logout command.
+	// +optional
+	ReplacementTimeout string
```

Comment:
It should be `*int32` (or `*int64`) - the field needs to be optional and nil value should mean the behavior we have today.

Reply:
@jsafrane just to add what @mittachaitu said , the other thing is if we use integer value, we have to convert it to string before passing it to UpdateISCSI method as iscsi-update operator accepts string args as a value, so anyway we have to do type conversion.

Reply:
This API is targeted to users, not iscsiadm. Users must see that an integer is expected. You'll get validation for free. Using OpenAPI, the validation can be even done on the client side.

How is the parameter converted internally and passed to iscsiadm is implementation detail.

Reply:
Agreed what you said, thanks,  we have changed to `*int32` and testing the changes right now.